packages/client/components/EstimateDimensionColumn.tsx

```diff
@@ -76,6 +77,10 @@ const EstimateDimensionColumn = (props: Props) => {
     if (submitting) return
     submitMutation()
     SetPokerSpectateMutation(atmosphere, {meetingId, isSpectating}, {onError, onCompleted})
+    const viewerHasVoted = !!scores.find(({userId}) => userId === viewerId)
+    if (viewerHasVoted) {
+      VoteForPokerStoryMutation(atmosphere, {meetingId, stageId}, {onError, onCompleted})
```

Comment:
If a user clicks "I don't vote", we shouldn't include them in the "Reveal Votes" progress count. There's a few ways of doing this:
1. Filter spectators in `scores` in `EstimateStage` [here](https://github.com/ParabolInc/parabol/blob/8685720f380ccd0081aa413c7d64fa8fb0cb91fe/packages/server/graphql/types/EstimateStage.ts#L177). This would mean grabbing all users every time we query `scores`, and we would need to run the query after every `SetPokerSpectateMutation`.
2. Filter on the client. This would add an extra burden to the client, and we could forget to filter elsewhere
3. My approach is removing the viewer's vote if they click "I don't vote", which means we don't need to think about filtering spectator scores. However, we do need to query `scores` on the client. Also, if the viewer votes for 5, clicks "I don't vote", clicks "Let me vote!", they won't see their vote for number 5. I'd say this is arguably the expected UX and keeps things clean logic wise.

Reply:
-1 I think the idea is right, but I think it should be part of `setPokerSpectate` mutation on server side and the optimistic updater.

Reply:
I was umming and arring about client vs server side here. It's more verbose server-side vs client, but I think you're right. If the next dev uses `SetPokerSpectateMutation ` and forgets to use `VoteForPokerStoryMutation`, they'll run into the same bug.